en/core-libraries/components/flash.rst

```diff
@@ -0,0 +1,78 @@
+Flash
+#####
+
+.. php:class:: FlashComponent(ComponentCollection $collection, array $config = [])
+
+FlashComponent provides a way to set one-time notification messages to be
+displayed after processing a form or acknowledging data. CakePHP refers to these
+messages as "flash messages". FlashComponent writes flash messages to
+``$_SESSION``, to be rendered in a View using
+:doc:`FlashHelper </core-libraries/helpers/flash>`.
+
+Setting Flash Messages
+======================
+
+FlashComponent provides two ways to set flash messages: its ``__call`` magic
+method and its ``set()`` method.  To furnish your application with verbosity,
+FlashComponent's ``__call`` magic method allows you use a method name that maps
+to an element located under the ``app/View/Elements/Flash`` directory. By
+convention, camelcased methods will map to the lowercased and underscored
+element name::
+
+    // Uses app/View/Elements/Flash/success.ctp
+    $this->Flash->success('This was successful');
+
+    // Uses app/View/Elements/Flash/great_success.ctp
+    $this->Flash->greatSuccess('This was greatly successful');
+
+Alternatively, to set a plain-text message without rendering an element, you can
+use the ``set()`` method::
+
+    $this->Flash->set('This is a message');
+
+FlashComponent's ``__call`` and ``set()`` methods optionally take a second
+parameter, an array of options:
+
+* ``key`` Defaults to 'flash'. The array key found under the 'Flash' key in
+  the session.
+* ``element`` Defaults to null, but will automatically be set when using the
+  ``__call()`` magic method. The element name to use for rendering.
+* ``params`` An optional array of keys/values to make available as variables
+  within an element.
+
+An example of using these options::
+
+    // In your Controller
+    $this->Flash->success('The user has been saved', [
+        'key' => 'positive',
+        'params' => [
+            'name' => $user['User']['name'],
+            'email' => $user['User']['email']
+        ]
+    ]);
+
+    // In your View
+    <?php echo $this->Flash->render('positive') ?>
+
+    <!-- In app/View/Elements/Flash/success.ctp -->
+    <div id="flash-<?php echo h($key) ?>" class="message-info success">
+        <?php echo h($message) ?>: <?php echo h($params['name']) ?>, <?php echo h($params['email']) ?>.
+    </div>
+
+Note that the parameter ``element`` will be always overridden while using ``__call()``.
```

Comment:
May be better the other way round:
If you use the parameter ``__call()``, ``element`` will be always overridden.

And shouldn't it be a "note" instead of just saying "Note..."?